src/main/java/uk/gov/hmcts/darts/arm/service/impl/AbstractArmBatchProcessResponseFiles.java

```diff
@@ -63,11 +69,13 @@
 
 @Slf4j
 @SuppressWarnings({"PMD.GodClass", "PMD.CyclomaticComplexity", "PMD.CouplingBetweenObjects"})
+@RequiredArgsConstructor
 public abstract class AbstractArmBatchProcessResponseFiles implements ArmResponseFilesProcessor {
 
     protected static final String UNABLE_TO_UPDATE_EOD = "Unable to update EOD";
     protected static final String CREATE_RECORD = "create_record";
     protected final ExternalObjectDirectoryRepository externalObjectDirectoryRepository;
+    protected final ObjectStateRecordRepository objectStateRecordRepository;
```

Comment:
The ObjectStateRecordRepository is only applicable to DETS and so should not be in this class. This is why there is a DETS version which specifically handles the code that is specific to DETS.

Reply:
Looking at the ticket and from my understanding we should update the OSR for both DARTs and DETs @hemantasharma1129 please can you confirm

Reply:
There isn't an OSR record for EOD's that are marked as DARTS. The OSR table is inserted by the migration team to insert the legacy data into modernised darts. EOD's with a manifest prefix of DARTS will all be generated by the modernised system darts-api and will be prefixed with DARTS. All the legacy data that is migrated will be prefixed by DETS. This is how we distinguish between the 2. Only the migrations team will insert a record into the OSR table.

Reply:
There is no OSR update for DARTS. I do agree the ticket should have made it clear that OSR update should only happen for DETS flow. Can we please update the code accordingly? Ticket will be updated now to add the DETS clause. @Ben-Edwards-cgi

Reply:
Thank you for confirming code updated and pushed :)